### PR TITLE
Add support for the document:deleteFields API action

### PR DESCRIPTION
**Pull request by Shiranuit**
## What does this PR do ?
This PR add the action `document:deleteFields` and the associated documentation and tests
⚠️ Depends on #632

### CODE REVIEW

File: src/controllers/Document.ts [hunk 1]
```diff
@@ -187,6 +187,48 @@ export class DocumentController extends BaseController {
...
+   *    - `silent` If true, then Kuzzle will not generate notifications
+   *    - `source` If true, the response will contain the updated document
+   *
+   * @returns The updated document
+   */
+  deleteFields(
+    index: string,
+    collection: string,
+    _id: string,
+    fields: Array<string>,
```
Comment (Aschen) [inline src/controllers/Document.ts:210]: Prefer the more concise `string[]`

**Review by Aschen — CHANGES_REQUESTED** [review]
You should add the `timeout` option from your other PR

**Review by MathieuVeber — APPROVED** [review]
LGTM :wink:

**Review by Aschen — APPROVED** [review]

File: doc/7/controllers/document/delete-fields/snippets/deleteFields.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+try {
+  const response = await kuzzle.document.deleteFields('nyc-open-data', 'yellow-taxi', 'some-id', ['bar'], {source: true});
```
Comment (Leodau) [inline doc/7/controllers/document/delete-fields/snippets/deleteFields.js:2]: Nit
```suggestion
  const response = await kuzzle.document.deleteFields('nyc-open-data', 'yellow-taxi', 'some-id', ['bar'], { source: true });
```

**Review by Leodau — APPROVED** [review]

**Pull request merged**